core/src/lib/output_formatter_resource.h

```diff
@@ -0,0 +1,148 @@
+ private:
+  /*
+   * Members
+   */
+  OutputFormatter* send_;
+  int indent_level_ = 0;
+
+ private:
+  /*
+   * Methods
+   */
+  bool requiresEscaping(const char* o);
+  void KeyMultipleStringsOnePerLineAddItem(const char* key,
+                                           const char* item,
+                                           bool as_comment,
+                                           bool quoted_strings);
+
+
+ public:
+  /*
+   * Methods
+   */
+  OutputFormatterResource(OutputFormatter* send, int indent_level = 0);
+  ~OutputFormatterResource();
+
+  OutputFormatter* GetOutputFormatter() { return send_; }
+
+  std::string GetKeyFormatString(bool inherited,
+                                 std::string baseformat = "%s = ");
+
+  void ResourceTypeStart(const char* name, bool as_comment = false);
+  void ResourceTypeEnd(const char* name, bool as_comment = false);
+
+  void ResourceStart(const char* name);
+  void ResourceEnd(const char* name);
+
+  void SubResourceStart(const char* name,
+                        bool as_comment = false,
+                        std::string baseformat = "%s {\n");
+  void SubResourceEnd(const char* name,
+                      bool as_comment = false,
+                      std::string baseformat = "}\n");
+
+  void KeyBool(const char* name, bool value, bool as_comment = false);
+
+  void KeySignedInt(const char* name, int64_t value, bool as_comment = false);
+  void KeyUnsignedInt(const char* name, int64_t value, bool as_comment = false);
+
+  void KeyString(const char* name, const char* value, bool as_comment = false)
+  {
+    KeyUnquotedString(name, value, as_comment);
+  }
+  void KeyString(const char* name, std::string value, bool as_comment = false)
+  {
+    KeyUnquotedString(name, value.c_str(), as_comment);
+  }
+  void KeyQuotedString(const char* name,
+                       const char* value,
+                       bool as_comment = false);
+  void KeyQuotedString(const char* name,
+                       std::string value,
+                       bool as_comment = false);
+  void KeyUnquotedString(const char* name,
+                         const char* value,
+                         bool as_comment = false);
+  void KeyUnquotedString(const char* name,
+                         std::string value,
+                         bool as_comment = false);
+
+  void KeyMultipleStringsInOneLine(const char* key,
+                                   alist* list,
+                                   bool as_comment = false,
+                                   bool quoted_strings = true);
+
+  void KeyMultipleStringsInOneLine(const char* key,
+                                   alist* list,
+                                   const char* (*GetValue)(void* item),
+                                   bool as_comment = false,
+                                   bool quoted_strings = true);
+
+  void KeyMultipleStringsOnePerLine(const char* key,
+                                    alist& list,
```

Comment:
Why alist as pointer here and alist* otherwise? 
Could you try to eliminate alist at all?